tests/kernel/interrupt/src/interrupt_offload.c

```diff
@@ -125,55 +132,70 @@ static void trigger_offload_interrupt(const bool real_irq, void *work)
 
 static void t_running(void *p1, void *p2, void *p3)
 {
-	while (1) {
+	k_sem_give(&sync_sem);
+
+	while (wait_for_end == 0) {
 		orig_t_keep_run = 1;
-		k_usleep(1);
+		k_usleep(150);
 	}
+
+	orig_t_keep_run = 1;
+	k_sem_give(&sync_sem);
 }
 
-static void run_test_offload(int case_type, int real_irq)
+static void init_env(int real_irq)
 {
 	static bool wq_already_start;
-	int thread_prio = 1;
-
-	TC_PRINT("case %d\n", case_type);
 
 	/* semaphore used to sync the end */
-	k_sem_init(&end_sem, 0, NUM_WORK);
+	k_sem_init(&sync_sem, 0, 1);
+	k_sem_init(&end_sem, 0, 1);
```

Comment:
The limit for `end_sem` must be equal to the `NUM_WORK`.

Reply:
You are right! Got it fixed, thanks!